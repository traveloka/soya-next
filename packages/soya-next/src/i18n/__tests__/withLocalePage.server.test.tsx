/**
 * @jest-environment node
 */
import setupTest from "./utils/withLocalePageSetup";
import type { IncomingMessage } from "http";

describe("withLocalePage - Server", () => {
  it("should add default locale, locale, and site locales to page props", async () => {
    const { commonProps, context, PageWithLocale } = setupTest();
    const props = await PageWithLocale.getInitialProps({
      ...commonProps,
      req: context as IncomingMessage,
    });
    expect(props).toMatchSnapshot();
  });
});
