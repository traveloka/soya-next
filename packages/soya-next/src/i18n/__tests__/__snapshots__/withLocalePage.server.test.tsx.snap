// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLocalePage - Server should add default locale, locale, and site locales to page props 1`] = `
Object {
  "defaultLocale": "id-id",
  "init": true,
  "locale": Object {
    "country": "id",
    "language": "id",
  },
  "siteLocales": Array [
    "id-id",
    "en-id",
  ],
}
`;
